### DISCUSSION TITLE
Small Contract Project for hire

### DISCUSSION THREAD

**Discussion by gamblw** [ImageMagick/ImageMagick#2507] [Consulting]
Hello!

First off, apologies if this is not the right place to post this. I saw in the older IM forum that there was a section for posting work for hire, so hopefully this fits within community rules.

I am looking to pay someone to build an IM script that will extract the handwriting from the image and return it back as a PNG with transparency. [Here is an example of an image that we would want to extract the handwriting from](https://media-kaliumllc.netdna-ssl.com/public/media/26507/optimized_1599147939-IMG_20200903_113522.jpg).

As you will see, there are other elements (e.g. desk in the background) that would also need to be removed to deliver just the handwriting.

I've looked at a few sample scripts that accomplished something similar however most of them added artifacts to the edges of the handwriting. We want to capture the handwriting as authentically as possible.

I don't have a set budget for the project and I'm open to either a per hour agreement or a specified project amount. 

If you have any questions please let me know. I can also be emailed at gamblw@gmail.com.

Thanks for your consideration,
Bill

**Comment by dlemstra** [comment]
At this moment we cannot change the categories because @GitHub is still working on that. We should be able to do that in the future and we will add a section for consulting then.

**Comment by fmw42** [comment]
Code for what platform? What version of ImageMagick?

**Reply by gamblw** [reply]
We are using ImageMagick 6.9.2-5 Q16 x86_64 2015-11-02 now, but I don't think there are any barriers to upgrading to a more current version (we should anyways).

We are calling it from RoR via mini_magick.

**Comment by fmw42** [comment]
Here is one way to do that in Unix syntax with ImageMagick. It uses connected components and an aggressive trim to extract the page and threshold it. The threshold image is then put in the alpha channel of the input to make it transparent.

```
thresh=70
area_percent=25
WxH=`convert handwriting.jpg -format "%wx%h" info:`
ww=`echo $WxH | cut -dx -f1`
hh=`echo $WxH | cut -dx -f2`
area_thresh=`convert handwriting.jpg -format "%[fx:$area_percent*$ww*$hh/100]" info:`
bbox=`convert handwriting.jpg -threshold $thresh% -type bilevel \
-define connected-components:exclude-header=true \
-define connected-components:verbose=true \
-define connected-components:area-threshold=area_thresh \
-define connected-components:mean-color=true \
-connected-components 8 null: | grep "gray(255)" | head -n 1 | awk '{print $2}'`
offx=`echo $bbox | cut -d+ -f2`
offy=`echo $bbox | cut -d+ -f3`
convert handwriting.jpg \
\( -clone 0 -threshold $thresh% -crop $bbox \
-define trim:percent-background=0% -background black -trim \
-background white -flatten -negate \) \
-alpha off -compose copy_opacity -composite \
handwriting_result.png

```


![handwriting_result](https://user-images.githubusercontent.com/13458218/92148902-13d71800-edd2-11ea-8b34-931a3b4c9b99.png)

**Reply by gamblw** [reply]
Wow Fred, I don't know what to say, this is awesome! I'm going to test it out a bunch more and will be back in touch.

**Comment by fmw42** [comment]
The threshold may need to be adjusted. You can also try in IM 7 using magick rather than convert and use -auto-threshold otsu to see if an automatic threshold will work. It is only implemented in IM 7 at 7.0.6-1 or higher. The aggressive trim is only available at 7.0.8-30 or higher or 6.9.10-30 or higher.

**Reply by gamblw** [reply]
Got it, I'll play around with that as well. Thanks again, you blew my mind!
